tests/MongoDB.Driver.Tests/Specifications/change-streams/ChangeStreamTestRunner.cs

```diff
@@ -301,6 +302,20 @@ private List<ChangeStreamDocument<BsonDocument>> ReadChangeStreamDocuments(IAsyn
             }
 
             return result;
+
+            IEnumerable<IEnumerable<ChangeStreamDocument<BsonDocument>>> ReliableMoveNext()
+            {
+                var stopwatch = Stopwatch.StartNew();
+                while (async ? cursor.MoveNextAsync().GetAwaiter().GetResult() : cursor.MoveNext())
+                {
+                    if (cursor.Current.Any() || stopwatch.Elapsed > TimeSpan.FromSeconds(10))
+                    {
+                        yield return cursor.Current;
+                    }
+
+                    Thread.Sleep(TimeSpan.FromMilliseconds(300)); // delay between attempts
```

Comment:
Nit: I think we should move `TimeSpan.FromMilliseconds(300)` and `TimeSpan.FromSeconds(10)` into named variables so they look less like magic numbers.

Reply:
I think a comment makes the same work, but I agree that 10 seconds could be described a bit too. I will do it.